File: server/FinancialMonitor.Api/FinancialMonitor.Api/Models/Transaction.cs
namespace FinancialMonitor.Api.Models;

public class Transaction
{
    public Guid transactionId { get; set; }

    public decimal amount { get; set; }

    public string currency { get; set; } = string.Empty;

    public TransactionStatus status { get; set; }

    public DateTime timestamp { get; set; }
}

public enum TransactionStatus
{
    Pending,
    Completed,
    Failed
}

File: client/src/types/transaction.ts
/**
 * Client-side transaction domain types.
 *
 * The shape mirrors the backend `TransactionResponse` DTO so that mock data
 * can later be swapped for real API data without changing the UI components.
 */

export type TransactionStatus = 'Pending' | 'Completed' | 'Failed'

export interface Transaction {
  transactionId: string
  amount: number
  currency: string
  status: TransactionStatus
  timestamp: string
}

export interface TransactionTotals {
  total: number
  pending: number
  completed: number
  failed: number
}
